src/indent.cpp

```diff
@@ -2171,6 +2171,7 @@ void indent_text(void)
             if (  pc->level == pc->brace_level
                && (  chunk_is_token(pc, CT_FPAREN_OPEN)
                   || chunk_is_token(pc, CT_SPAREN_OPEN)
+                  || (language_is_set(LANG_CPP) && chunk_is_token(pc, CT_SQUARE_OPEN))
```

Comment:
I'm not sure the language check is wanted here. At least, I think it should apply to C, also?

Reply:
Yes, it is.

Reply:
Let's say I have the code:

```c
int x[5 +
7];
```

This is valid in both C and C++. But with the language check, it will be formatted differently if formatted as C vs. as C++. I don't understand why that is desired?

```
$ echo $'int x[5 +\n7];' | ./uncrustify -lC -c ../tests/config/Issue_2045.cfg
Parsing: 14 bytes (14 chars) from stdin as language C
int x[5 +
7];

$ echo $'int x[5 +\n7];' | ./uncrustify -lCPP -c ../tests/config/Issue_2045.cfg
Parsing: 14 bytes (14 chars) from stdin as language CPP
int x[5 +
        7];
```

Reply:
The test was introduced because of:
`oc/50514-indent_boxed.m`
```.diff
@@ -28,14 +28,14 @@
     ];
     NSMutableArray* f = [NSMutableArray arrayWithArray:@[@[@1], @[@2], @[@3]]];
     NSMutableArray* g = [NSMutableArray arrayWithArray:@[
-           @[@1], @[@2], @[@3]
+               @[@1], @[@2], @[@3]
     ]];

```
I am not an expert for oc.
Are the changes OK?

Reply:
Hmm... I don't do OC either... but wouldn't OC++ have the same issue?

Assuming that OC is not sensitive to the indent, I would think that this is either in line with what #2045 is requesting, or indicates that the fix as currently proposed is inadequate.

Reply:
Okay... on closer consideration, I think the PR in its current state is inadequate. AFAICT, for some reason, `indent_continue` is being applied *twice* in OC blocks. This makes me wonder if there is actually code that can go away with this change...

Reply:
Now I am even *more* confused.

Consider:
```C++
auto x = y[
7];
```
IIUC, this *ought* to be getting `indent_continue` applied *twice*¹... as indeed is the case with:
```c++
auto x = foo(
7);
```
...and in which case I believe the changes in the OC tests are correct. Indeed I am starting to think they *are* correct, but I guess that comes down to, do we want to apply `indent_continue` to *all* `[`s, including "special" OC ones? Or do we exclude the "special" OC ones?

If the former, I believe the code is (for the OC tests) doing the right thing, and the test expected output should be adjusted. If the latter, I believe the correct fix is:
```diff
-(language_is_set(LANG_CPP) && chunk_is_token(pc, CT_SQUARE_OPEN))
+(chunk_is_token(pc, CT_SQUARE_OPEN) && pc->parent_type != CT_OS_MSG)
```

At any rate, I think this PR is still buggy somehow. Please consider the first test case in this comment; do you agree it should have `indent_continue` applied twice? If so, that is still not working.

(¹ `use_indent_continue_only_once=false`)